components/add_new_cv_form.tsx

```diff
@@ -86,27 +93,88 @@ export default function AddNewCvForm({ id }: Props) {
     }
 
     delete updatedCv.projects;
-    delete updatedCv.technical_skills;
+    delete updatedCv.certifications;
+    delete updatedCv.education;
 
     return supabase.from("cv").upsert(updatedCv).select();
   }
 
-  async function upsertSkills(values: any, cvData: any) {
-    values.cv_id = cvData[0].id;
-    return supabase.from("technical_skills").upsert(values);
+  async function upsertCertifications(certifications: any, cvId: string) {
+    if (certifications.length === 0) {
+      return await supabase.from("certifications").delete().eq("cv_id", cvId);
+    }
+
+    const updatedCertifications = certifications.map((certification: any) => ({
+      ...certification,
+      cv_id: cvId,
+    }));
+    return supabase.from("certifications").upsert(updatedCertifications);
+  }
+
+  async function upsertEducation(education: any, cvId: string) {
+    education.cv_id = cvId;
+    return supabase.from("education").upsert(education);
+  }
+
+  async function upsertProjects(projects: any, cvId: string) {
+    if (projects.length === 0) {
+      return await supabase.from("projects").delete().eq("cv_id", cvId);
+    }
+
+    const updatedProjects = projects.map((project: any) => {
+      const startDate = new Date(project.date_start);
+      const endDate = new Date(project.date_end);
+      const formattedStartDate = `${startDate.getDay() + 1}.${
+        startDate.getMonth() + 1
+      }.${startDate.getFullYear()}`;
+      const formattedEndDate = `${endDate.getDay() + 1}.${
+        endDate.getMonth() + 1
+      }.${endDate.getFullYear()}`;
+
+      return {
+        ...project,
+        cv_id: cvId,
+        date_start: formattedStartDate,
+        date_end: formattedEndDate,
```

Comment:
Usually you want to store the date on the server as a UTC timestamp and format only for display reasons.

Reply:
Yeah, definitely we should do this from the start to avoid inaccurate data. I know we're keeping only month and year so it's not a big of a deal, but let's make database data consistent.

Reply:
👍